backend/spec/performance/invoice_list_query_performance_spec.rb

```
@@ -0,0 +1,170 @@
+        times << time
+      end
+
+      # With proper indexing, query time should not increase linearly
+      # Even with 4x more data, time should not increase significantly
+      expect(times.last).to be < (times.first * 2)
+    end
+
+    it 'efficiently handles the exact TRPC query pattern' do
+      # Simulate the exact query from frontend/trpc/routes/invoices.ts
+      query_execution_plan = ActiveRecord::Base.connection.execute(<<~SQL).to_a
+        EXPLAIN (ANALYZE, BUFFERS, FORMAT JSON) 
```

RuboCop (1.75.5)

[convention] Trailing whitespace detected.

(Layout/TrailingWhitespace)